lib-es5/utils/index.js

```diff
@@ -222,71 +222,117 @@ function process_if(ifValue) {
-      if (isEmpty(text) && resource_type !== "text") {
-        if (noPublicId) {
-          throw "Must supply public_id for resource_type layer_parameter";
-        }
-        if (resource_type === "subtitles") {
-          style = textStyle(layer);
-        }
-      } else {
-        resource_type = "text";
-        type = null;
-        // type is ignored for text layers
-        style = textStyle(layer);
-        if (!isEmpty(text)) {
-          var noStyle = isEmpty(style);
-          if (!(noPublicId || noStyle) || noPublicId && noStyle) {
-            throw "Must supply either style parameters or a public_id when providing text parameter in a text overlay/underlay";
-          }
-          var re = /\$\([a-zA-Z]\w*\)/g;
-          var start = 0;
-          var textSource = smart_escape(decodeURIComponent(text), /[,\/]/g);
-          text = "";
-          for (var res = re.exec(textSource); res; res = re.exec(textSource)) {
-            text += smart_escape(textSource.slice(start, res.index));
-            text += res[0];
-            start = res.index + res[0].length;
-          }
-          text += encodeURIComponent(textSource.slice(start));
-        }
-      }
-      if (resource_type !== "image") {
-        components.push(resource_type);
-      }
-      if (type !== "upload") {
-        components.push(type);
-      }
-      components.push(style);
+      return layer;
+    }
+
+    layer = {
+      url: layerUrl,
+      type: 'fetch'
+    };
+
+    if (resourceType) {
+      layer.resource_type = resourceType;
+    }
+  }
+
+  if (typeof layer !== 'object') {
+    return layer;
+  }
+
+  var _layer = layer,
+      resource_type = _layer.resource_type,
+      text = _layer.text,
+      type = _layer.type,
+      public_id = _layer.public_id,
+      format = _layer.format,
+      fetchUrl = _layer.url;
+
+  var components = [];
+
+  if (!isEmpty(text) && isEmpty(resource_type)) {
+    resource_type = 'text';
+  }
+
+  if (!isEmpty(fetchUrl) && isEmpty(type)) {
+    type = 'fetch';
+  }
+
+  if (!isEmpty(public_id) && !isEmpty(format)) {
+    public_id = `${public_id}.${format}`;
+  }
+
+  if (isEmpty(public_id) && resource_type !== 'text' && type !== 'fetch') {
+    throw new Error('Must supply public_id for non-text overlay');
+  }
+
+  if (!isEmpty(resource_type) && resource_type !== 'image') {
+    components.push(resource_type);
+  }
+
+  if (!isEmpty(type) && type !== 'upload') {
+    components.push(type);
+  }
+
+  if (resource_type === 'text' || resource_type === 'subtitles') {
+    if (isEmpty(public_id) && isEmpty(text)) {
+      throw new Error('Must supply either text or public_in in overlay');
+    }
+
+    var textOptions = textStyle(layer);
+
+    if (!isEmpty(textOptions)) {
+      components.push(textOptions);
+    }
+
+    if (!isEmpty(public_id)) {
+      public_id = public_id.replace('/', ':');
       components.push(public_id);
+    }
+
+    if (!isEmpty(text)) {
+      // const pattern = /(\$\([a-zA-Z]\w+\))/g;
```

Comment:
Where this commented out code is coming from? is it in use? are there tests covering this functionality?

Reply:
Left it behind by mistake, already removed it.